.github/workflows/test_and_release.yml

```diff
@@ -0,0 +1,123 @@
+name: tests
```

Comment:
name: Test and Release

Could you please go through jobs and steps and make them canonical.